### PR TITLE
[ODS-4445] Remove usage of IConfigValueProvider

### PR DESCRIPTION
**Pull request by BhanuprakashThunga**

### CODE REVIEW

File: Application/EdFi.Admin.DataAccess/Utils/DefaultApplicationCreator.cs [hunk 1]
```diff
@@ -11,20 +11,21 @@
...
     public class DefaultApplicationCreator : IDefaultApplicationCreator
     {
-        private readonly IConfigValueProvider _configValueProvider;
+        private readonly IConfiguration _configuration;
         private readonly IUsersContextFactory _usersContextFactory;
 
         public DefaultApplicationCreator(
             IUsersContextFactory usersContextFactory,
-            IConfigValueProvider configValueProvider)
+            IConfiguration configuration)
```
Comment (jamessmckay) [inline Application/EdFi.Admin.DataAccess/Utils/DefaultApplicationCreator.cs:25]: This breaks the net usage of this file. What you should have done is copied the file over, and wrapped it in NETSTANDARD and made this change. The original file should have been renamed to net.cs and wrapped in NETFRAMEWORK

File: Application/EdFi.Common/Configuration/IConfigValueProvider.cs [hunk 2]
```diff
@@ -1,12 +1,15 @@
 // SPDX-License-Identifier: Apache-2.0
+#if NETFRAMEWORK
```
Comment (jamessmckay) [inline Application/EdFi.Common/Configuration/IConfigValueProvider.cs:2]: this is in the wrong spot it should be above the namespace

File: Application/EdFi.Ods.Common/Configuration/DefaultPageSizeLimitProvider.cs [hunk 3]
```diff
@@ -1,23 +1,22 @@
+#if NETSTANDARD
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Common/Configuration/DefaultPageSizeLimitProvider.cs:1]: this is in the wrong spot should be above the namespace.

File: Application/EdFi.Ods.Sandbox/Provisioners/PostgresSandboxProvisioner.cs [hunk 4]
```diff
@@ -10,6 +10,7 @@
 using EdFi.Admin.DataAccess.Utils;
 using EdFi.Ods.Common.Configuration;
 using log4net;
+using Microsoft.Extensions.Configuration;
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Sandbox/Provisioners/PostgresSandboxProvisioner.cs:13]: same problem as above, needed to make a copy and updated.

File: Application/EdFi.Ods.Sandbox/Provisioners/SandboxProvisionerBase.cs [hunk 5]
```diff
@@ -11,23 +11,24 @@
...
 using EdFi.Ods.Common.Configuration;
 using EdFi.Ods.Common.Extensions;
+using Microsoft.Extensions.Configuration;
 
 namespace EdFi.Ods.Sandbox.Provisioners
 {
     public abstract class SandboxProvisionerBase : ISandboxProvisioner
     {
-        private readonly IConfigValueProvider _configValueProvider;
+        private readonly IConfiguration _configuration;
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Sandbox/Provisioners/SandboxProvisionerBase.cs:20]: same as above make an independent copy

File: Application/EdFi.Ods.Sandbox/Provisioners/SqlServerSandboxProvisioner.cs [hunk 6]
```diff
@@ -13,16 +13,17 @@
...
+using Microsoft.Extensions.Configuration;
 
 namespace EdFi.Ods.Sandbox.Provisioners
 {
     public class SqlServerSandboxProvisioner : SandboxProvisionerBase
     {
         private readonly ILog _logger = LogManager.GetLogger(typeof(SqlServerSandboxProvisioner));
 
-        public SqlServerSandboxProvisioner(IConfigValueProvider configValueProvider,
+        public SqlServerSandboxProvisioner(IConfiguration configuration,
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Sandbox/Provisioners/SqlServerSandboxProvisioner.cs:24]: same as above make an independent copy

**Review by jamessmckay — CHANGES_REQUESTED** [review]

File: Application/EdFi.Admin.DataAccess/Utils/DefaultApplicationCreator.cs (hunk 1 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Admin.DataAccess/Utils/DefaultApplicationCreator.cs:26]: Created new file and wrapped inside NETFRAMEWORK

File: Application/EdFi.Common/Configuration/IConfigValueProvider.cs (hunk 2 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Common/Configuration/IConfigValueProvider.cs]: Moved above namespace

File: Application/EdFi.Ods.Common/Configuration/DefaultPageSizeLimitProvider.cs (hunk 3 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Common/Configuration/DefaultPageSizeLimitProvider.cs]: Moved above namespace

File: Application/EdFi.Ods.Sandbox/Provisioners/PostgresSandboxProvisioner.cs (hunk 4 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Sandbox/Provisioners/PostgresSandboxProvisioner.cs:14]: Created new file and wrapped inside NETFRAMEWORK

File: Application/EdFi.Ods.Sandbox/Provisioners/SandboxProvisionerBase.cs (hunk 5 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Sandbox/Provisioners/SandboxProvisionerBase.cs:21]: Created new file and wrapped inside NETFRAMEWORK

File: Application/EdFi.Ods.Sandbox/Provisioners/SqlServerSandboxProvisioner.cs (hunk 6 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Sandbox/Provisioners/SqlServerSandboxProvisioner.cs:25]: Creadted new file and wrapped inside NETFRAMEWORK

File: Application/EdFi.Admin.DataAccess/Utils/DefaultApplicationCreator.net..cs [hunk 7]
```diff
@@ -0,0 +1,95 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+#if NETFRAMEWORK
```
Comment (jamessmckay) [inline Application/EdFi.Admin.DataAccess/Utils/DefaultApplicationCreator.net..cs:6]: fix the name on this file please. it should be `DefaultApplicationCreate.net.cs` not `DefaultApplicationCreate.net..cs`

File: Application/EdFi.Ods.Common/Configuration/DefaultPageSizeLimitProvider.cs [hunk 8]
```diff
@@ -3,21 +3,19 @@
...
+        private readonly IConfiguration _configuration;
+        public DefaultPageSizeLimitProvider(IConfiguration configuration)
         {
-            _configValueProvider = configValueProvider;
-        }
-
-        public int GetDefaultPageSizeLimit() => int.TryParse(_configValueProvider.GetValue("defaultPageSizeLimit"), out int defaultPageSizeLimit)
+            _configuration = configuration;
+       }
+        public int GetDefaultPageSizeLimit() => int.TryParse(_configuration.GetSection("defaultPageSizeLimit").Value, out int defaultPageSizeLimit)
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Common/Configuration/DefaultPageSizeLimitProvider.cs:18]: formatting issue here. this needs to be fixed please. also fix the braces , and there should be a blank line between the statements

File: Application/EdFi.Ods.Sandbox/Provisioners/PostgresSandboxProvisioner.cs (hunk 4 above)
Comment (jamessmckay) [reply Application/EdFi.Ods.Sandbox/Provisioners/PostgresSandboxProvisioner.cs:14]: as a fyi in the future, we would normally create a new file for the netcoreapp changes. and modify the original file to be netframework.

File: Application/EdFi.Ods.Sandbox/Provisioners/SandboxProvisionerBase.cs [hunk 9]
```diff
@@ -11,23 +12,24 @@
...
         private readonly IConfigConnectionStringsProvider _connectionStringsProvider;
         protected readonly IDatabaseNameBuilder _databaseNameBuilder;
-
-        protected SandboxProvisionerBase(IConfigValueProvider configValueProvider,
+
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Sandbox/Provisioners/SandboxProvisionerBase.cs:24]: remove blank spaces please.

**Review by jamessmckay — CHANGES_REQUESTED** [review]

File: Application/EdFi.Ods.Common/Configuration/DefaultPageSizeLimitProvider.cs (hunk 8 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Common/Configuration/DefaultPageSizeLimitProvider.cs:20]: Installed resharper and formatted the code

File: Application/EdFi.Ods.Sandbox/Provisioners/SandboxProvisionerBase.cs (hunk 9 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Sandbox/Provisioners/SandboxProvisionerBase.cs:24]: Blank space should be there right, There is only one space

**Review by jamessmckay — APPROVED** [review]

**Pull request merged**